src/task2/connection.py

```diff
@@ -0,0 +1,129 @@
+import socket
+import enum
+import json
+import os
+
+encode_compact = json.JSONEncoder(separators=(",", ":")).encode
+
+
+class Action(enum.Enum):
+    LOGIN = 1
+    LOGOUT = 2
+    MOVE = 3
+    UPGRADE = 4
+    TURN = 5
+    PLAYER = 6
+    GAMES = 7
+    MAP = 10
+
+
+class Result(enum.Enum):
+    OKEY = 0
+    BAD_COMMAND = 1
+    RESOURCE_NOT_FOUND = 2
+    ACCESS_DENIED = 3
+    INAPPROPRIATE_GAME_STATE = 4
+    TIMEOUT = 5
+    INTERNAL_SERVER_ERROR = 500
+
+
+class GameConnectionError(Exception):
+    def __init__(self, code, error):
+        self.code = code
+        self.error = error
+        super().__init__(": ".join((code.name, error)))
+
+
+class ResponseError(GameConnectionError):
+    pass
+
+
+class Connector:
+    int_size = 4
+    encoding = "utf-8"
+
+    def __init__(self, address: "(SERVER_ADDRESS, SERVER_PORT)" = None):
+        directory = os.path.dirname(os.path.abspath(__file__))
+        if address is None:
+            with open(directory + r"\server_config.json") as json_data:
```

Comment:
it's not crossplatform
better use `os.path.join` to solve this problem

Reply:
was fixed